src/realm/parser/driver.cpp

```diff
@@ -1659,6 +1660,12 @@ Query Table::query(const std::string& query_string, query_parser::Arguments& arg
     return driver.result->visit(&driver).set_ordering(driver.ordering->visit(&driver));
 }
 
+Query Class::get_query(const std::string& query_string, query_parser::Arguments& args,
```

Comment:
Its a bit weird having this here rather than in class.cpp. But I guess that matches `Table::query()`, so 🤷.

I suspect that this should just be inlined into `class.hpp` anyway.

Reply:
This is to avoid linking in the query parser in case the SDK is not using it.

Reply:
I figured. But if you put it inline in the header, you don't have that problem. It is simpler for this method because it just forwards to `Table::get_query()`. So if it is inline, the compiler only emits the code to call _that_ method if `Class::get_query()` is actually called by the SDK. If you want it to be out-of-line, then I agree, it does need to be in this file.